docroot/profiles/custom/cgov_site/modules/custom/cgov_list/cgov_list.module

```diff
@@ -78,18 +78,39 @@ function format_bytes(int $bytes, int $precision = 2) {
 function cgov_list_allowed_values_function(
   FieldStorageConfig $definition,
   ContentEntityInterface $entity = NULL,
-  $cacheable
+  &$cacheable
 ) {
   // This initially queried the view modes for paragraphs; however,
   // managed lists should only use a subset of the list item display
   // modes. So we are keeping it like this if the future requires it,
   // also it is much easier to change post go-live.
-  $options = [
-    'list_item_title' => 'Compact',
-    'list_item_title_desc' => 'Title and Description',
-    'list_item_title_desc_image' => 'Title, Description, and Image',
-      'list_item_title_desc_image_alternating' => 'Title, Description, and Alternating Image',
-  ];
+  $request = \Drupal::requestStack()->getCurrentRequest();
+  $node = \Drupal::routeMatch()->getParameter('node');
+  $route_name = \Drupal::routeMatch()->getRouteName();
+
+  if ($node->field_mlp_page_style->value === "ncids_default") {
+    $options = [
+      'ncids_list_item_title_desc' => 'Title and Description',
+      'ncids_list_item_title_desc_image' => 'Title, Description, and Image',
+    ];
+  }
+  elseif (in_array($route_name, ['node.add'])) {
```

Comment:
You should check first if this is the `node.add` route rather than fetching the current entity. This will make it easier to read. I.E. If this is node.add then do X, else, then let's see what the field items are set to.

Now, also, what happens when I change the style? This is going to get set as the options for this node once. So we need to be doubly sure that we are not just testing the happy path when testing this, as well as making sure adding old template works.